### PR TITLE
Standardize logging

### PR DESCRIPTION
**Pull request by labrys**
- [ ] PR is based on the DEVELOP branch
- [ ] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [ ] Read the [contribution guide](https://github.com/pymedusa/Medusa/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: medusa/name_parser/parser.py [hunk 1]
```diff
@@ -117,35 +119,52 @@ def _parse_string(self, name):
...
+                    log.warning(
+                        'Indexer exception: {indexer_api.name}: {error}',
+                        {'indexer_api': indexer_api, 'error': error.message}
+                    )
                     episode_numbers = []
 
             for episode_number in episode_numbers:
                 s = season_number
-                e = episode_number
+                error = episode_number
```
Comment (ratoaq2) [inline medusa/name_parser/parser.py]: Why error?

File: medusa/name_parser/parser.py [hunk 2]
```diff
@@ -117,35 +119,52 @@ def _parse_string(self, name):
...
                     episode_numbers = []
-                except IndexerError as e:
-                    logger.warning('Unable to contact {indexer_api.name}: {e}', indexer_api=indexer_api, e=e.message)
+                except IndexerError as error:
+                    log.warning(
+                        'Unable to contact {indexer_api.name}: {error}',
+                        {'indexer_api': indexer_api, 'error': error.message}
+                    )
                     episode_numbers = []
-                except IndexerException as e:
```
Comment (fernandog) [inline medusa/name_parser/parser.py]: are we changing default `as e`  as we agreed in the past?

Comment (labrys) [inline medusa/name_parser/parser.py]: We agreed in the past that single letter variables like `e` should be avoided.

File: medusa/name_parser/parser.py (hunk 1 above)
Comment (labrys) [inline medusa/name_parser/parser.py]: messed up refactor... was refactoring `Exception as e` and somehow that got changed too

File: medusa/name_parser/parser.py (hunk 2 above)
Comment (fernandog) [inline medusa/name_parser/parser.py]: where is documented?

File: medusa/name_parser/parser.py (hunk 1 above)
Comment (labrys) [inline medusa/name_parser/parser.py]: `e = episode_number` overwrites the existing `e` in the scope from `Exception as e` so auto-refactor refactored both

File: medusa/name_parser/parser.py (hunk 2 above)
Comment (labrys) [inline medusa/name_parser/parser.py]: @fernandog my second comment to @ratoaq2 is exactly why names like `e` are bad...  its too easy to end up clobbering an existing variable and its not descriptive of the content... just like `e` for episode is bad too.

File: medusa/name_parser/parser.py [hunk 3]
```diff
@@ -156,53 +175,59 @@ def _parse_string(self, name):
...
-                new_episode_numbers.append(e)
-                new_season_numbers.append(s)
+                new_episode_numbers.append(episode)
+                new_season_numbers.append(season)
 
         # need to do a quick sanity check heregex.  It's possible that we now have episodes
         # from more than one season (by tvdb numbering), and this is just too much
         # for the application, so we'd need to flag it.
-        new_season_numbers = list(set(new_season_numbers))  # remove duplicates
+        new_season_numbers = sorted(set(new_season_numbers))  # remove duplicates
```
Comment (p0psicles) [inline medusa/name_parser/parser.py]: Okay I checked this. Was concerned at first but it returns a list. So seems safe.

File: medusa/name_parser/parser.py [hunk 4]
```diff
@@ -264,11 +293,13 @@ def assert_supported(result):
         :type result: ParseResult
         """
         if not result.show:
-            raise InvalidShowException('Unable to match {result.original_name} to a show in your database. '
+            raise InvalidShowException('Unable to match {result.original_name} to a series in your database. '
                                        'Parser result: {result}'.format(result=result))
```
Comment (p0psicles) [inline medusa/name_parser/parser.py]: Nice

File: medusa/name_parser/rules/rules.py [hunk 5]
```diff
@@ -1220,8 +1222,8 @@ def when(self, matches, context):
                     to_remove.extend(matches.named('title', predicate=lambda match: match.value != values[0].value))
                     continue
 
-                logger.info(u"Guessed more than one '%s' for '%s': %s",
-                            name, matches.input_string, u','.join(unique_values), exc_info=False)
+                log.info(u"Guessed more than one '%s' for '%s': %s",
```
Comment (p0psicles) [inline medusa/name_parser/rules/rules.py]: Let's use {0} for Positional argument substitution for Medusa code okay? The we limit the amount of log patterns.

Comment (labrys) [inline medusa/name_parser/rules/rules.py]: @p0psicles I left this unchanged intentionally, as there was only a single log line I figured that importing the `BraceAdapter` may not be necessary.  Thoughts?

Comment (p0psicles) [inline medusa/name_parser/rules/rules.py]: Okay I can see that.

**Pull request merged**